### PR TITLE
Reword "using the API" and related steps

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) Change ``Given using API version`` to ``Given using OCS API version``
2) Change other ``using the API`` to more specific ``using the xxx API``

## Motivation and Context
There are OCS API versions, and within that a sharing API version. That makes this step confusing:
```
Given using API version "1"
```
as it is not clear what "API" it is.

Similarly other test step text that says ``using the API`` can sometimes be unclear, because maybe there can be multiple ways to do something using an API endpoint(s). So be specific about the various APIs that exist - ``sharing API`` ``trashbin API`` ``webDAV API`` and so on.

## How Has This Been Tested?
- CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [x] Code changes
- [x] Acceptance tests added

## Open tasks:
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by phil-davis** [comment]
Backport ``stable10`` #32129

File: tests/acceptance/features/apiProvisioning-v1/addGroup.feature [hunk 1]
```diff
@@ -5,10 +5,10 @@ I want to be able to add groups
 So that I can more easily manage access to resources by groups rather than individual users
 
 	Background:
-		Given using API version "1"
+		Given using OCS API version "1"
 
 	Scenario Outline: admin creates a group
-		Given the administrator sends a group creation request for group "<group_id>" using the API
+		Given the administrator sends a group creation request for group "<group_id>" using the provisioning API
```
Comment (individual-it) [inline tests/acceptance/features/apiProvisioning-v1/addGroup.feature]: this is a WHEN step

File: tests/acceptance/features/apiProvisioning-v1/addGroup.feature [hunk 2]
```diff
@@ -51,7 +51,7 @@ So that I can more easily manage access to resources by groups rather than indiv
 		Given user "subadmin" has been created
 		And group "new-group" has been created
 		And user "subadmin" has been made a subadmin of group "new-group"
-		And user "subadmin" sends HTTP method "POST" to API endpoint "/cloud/groups" with body
+		And user "subadmin" sends HTTP method "POST" to OCS API endpoint "/cloud/groups" with body
```
Comment (individual-it) [inline tests/acceptance/features/apiProvisioning-v1/addGroup.feature]: this is a WHEN step

File: tests/acceptance/features/apiMain/transfer-ownership.feature [hunk 3]
```diff
@@ -297,7 +297,7 @@ Feature: transfer-ownership
 	Scenario: transferring ownership fails with empty files
 		Given user "user0" has been created
 		And user "user1" has been created
-		And user "user0" deletes everything from folder "/" using the API
+		And user "user0" deletes everything from folder "/" using the WebDAV API
```
Comment (individual-it) [inline tests/acceptance/features/apiMain/transfer-ownership.feature]: WHEN step or do not use 'using the WebDAV API'

File: tests/acceptance/features/apiMain/transfer-ownership.feature [hunk 4]
```diff
@@ -184,9 +184,9 @@ Feature: transfer-ownership
 		And user "user1" has been created
 		And user "user2" has been created
 		And user "user0" has created a folder "/test"
-		And user "user0" uploads file "data/textfile.txt" to "/test/somefile.txt" using the API
+		And user "user0" has uploaded file "data/textfile.txt" to "/test/somefile.txt"
 		And user "user0" has shared folder "/test" with user "user2" with permissions 31
-		And user "user1" creates a share using the API with settings
+		And user "user1" creates a share using the sharing API with settings
```
Comment (individual-it) [inline tests/acceptance/features/apiMain/transfer-ownership.feature]: WHEN step or do not use 'using the WebDAV API'

**Review by individual-it — APPROVED** [review]
a couple of places where "When" steps are used as "Given", can also be fixed in a separate PR

**Pull request merged**